exercise/api/views.py

```diff
@@ -222,6 +226,77 @@ def create(self, request, exercise_id, version):
 
             return Response(status=status.HTTP_201_CREATED)
 
+    @action(
+        detail=False,
+        url_path='submit',
+        url_name='submit',
+        methods=['post'],
+    )
+    def submit(self, request, *args, **kwargs):
```

Comment:
Should this deprecate `create` or can they share the code? How possible you think it would be to remove the submission django view in favour of this (e.g. in January release)? My fear is that these 3 different code paths have some differences, or if one is fixed then two others will be broken.

Reply:
The `create` method corresponds to a very different use case, so I didn't want to touch it. I don't know if anyone really ever uses that endpoint, though. The `create` method creates manually graded submissions to the database. The submission is not sent to the grader. It also looks suspicious that the code checks if the teacher is allowed to make submissions (based on deadlines and such even though staff is not affected by them), but the submission is created for other users based on the POST data.

I thought that I should create a new endpoint since the the `create` method has very little to do with the new use case (students submitting normal submissions for grading).

Removing the normal Django submission view should be possible. The API already supports session-based authentication that is needed in the normal web. The HTTP responses in the API are different than in the normal views, hence rendering HTML for the browser requires work.

Reply:
> The HTTP responses in the API are different than in the normal views, hence rendering HTML for the browser requires work.
That is true. I guess we could look into that for the next release. Of course, we could use a common function for both with very small presentation differences.

> The create method creates manually graded submissions to the database
Ouh yes. It was added to add minimalistic support for manually grading exercises. I think it's good in that way as it's be besic "add" endpoint.